lib/processors/bundlers/flexChangesBundler.js

```diff
@@ -20,9 +20,10 @@ import {createResource} from "@ui5/fs/resourceFactory";
  * @param {string} parameters.options.pathPrefix Prefix for bundle path
  * @param {string} parameters.options.hasFlexBundleVersion true if minUI5Version >= 1.73 than
  *															create flexibility-bundle.json
+ * @param {object} parameters.existingFlexBundle
  * @returns {Promise<@ui5/fs/Resource[]>} Promise resolving with flex changes bundle resources
  */
-export default function({resources, options: {pathPrefix, hasFlexBundleVersion}}) {
+export default function({resources, options: {pathPrefix, hasFlexBundleVersion}, existingFlexBundle = {}}) {
```

Comment:
I'm not sure what's the best way of providing the existing bundle. As the processor already knows about @ui5/fs/Resource, it could receive it instead of directly getting the string, which would include the path information, but that's most likely not relevant.

In addition, I'm not sure whether the new parameter should be top-level (as it is right now) or within the `options`.

@RandomByte

Reply:
Indeed the processor could also receive a Resource instance, which would make the signature a little more consistent (receiving all files in the form of resources). But I'm fine with passing the parsed JSON too.

Consider whether the resource "path" or other file system information of the existing bundle might be important in the future. And also how someone might want to re-use this processor outside of UI5 Tooling.

On the matter of whether to put the new parameter into `options` I'm not quite sure. It doesn't really feel like an option to control the processor's behavior, but rather like general data. Maybe similar to the [manifestCrator's `libraryResource`](https://github.com/SAP/ui5-builder/blob/b84d3e9eb2a8db0f6d50388c6acccdc5a5f8db03/lib/processors/manifestCreator.js#L629), which stands outside the `options`

Reply:
The idea behind the new parameter is to provide the existing content of flexibility-bundle.json.
The path information will not be needed, because it's already fix if option.hasFlexBundleVersion is true.
I placed it as last parameter because the priority to fill the parameter is the lowest.
For reuse purpose I decided to use the object data type, if for later development the existing flexibility-bundle.json is fetched from other sources.